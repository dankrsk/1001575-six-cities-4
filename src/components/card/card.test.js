import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../card/card.jsx';

const offer = {
  id: 1,
  type: `House`,
  title: `Beautiful & luxurious apartment at great location`,
  price: 100,
  rating: 1,
  isPremium: true,
  photo: `http://cdn.home-designing.com/wp-content/uploads/2013/11/1-Contemporary-architecture.jpg`,
};

describe(`Snapshots for Card`, () => {
  it(`Card component should render 1 card for mock offer`, () => {
    const tree = renderer
      .create(<Card
        offer={offer}
        onCardMouseOver={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
