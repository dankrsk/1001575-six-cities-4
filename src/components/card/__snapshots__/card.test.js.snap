// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots for Card Card component 1`] = `
<article
  className="place-card cities__place-card"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="place-card__image-wrapper cities__image-wrapper"
  >
    <a>
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="http://cdn.home-designing.com/wp-content/uploads/2013/11/1-Contemporary-architecture.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          100
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button place-card__bookmark-button--active"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "20%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        Beautiful & luxurious apartment at great location
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      House
    </p>
  </div>
</article>
`;
